### DISCUSSION TITLE
Paging에서 제공하는 정보에 대해

### DISCUSSION THREAD

**Discussion by rawfishthelgh** [woowacourse-teams/2023-pium#141] [뜨거운 토론]
현재 반려식물 관리 히스토리 조회는 다음과 같은 방식으로 이루어지고 있습니다.

**URL**
`GET``/history?petPlantId=1&page=1&size=10`

**RESPONSE**
```json
{
        "page": 1,
	"size": 10,
	"elementSize": 100,
	"hasNext": true,
	"waterDateList": [
			"2023-07-01",
			"2023-07-02",
			"2023-07-03"
	]
}
```

### 1. URL에 이미 존재하는 page, size 정보를 body에서 또 제공할 필요가 있는가?
현재 url과 responsebody에 존재하는 page, size 정보는 완전히 같은 정보입니다. 이미 존재하고 있는 정보를 다시 한 번 제공하는것이 데이터의 중복이 아닌가라는 의문이 들었습니다. 이 부분에 대한 의견을 듣고 싶습니다ㅎㅎ

### 2. 페이지의 인덱스를 0부터 시작할까 1부터 시작할까?
현재 페이징 처리를 구현하는 시스템은 페이징의 시작 인덱스를 0을 기준으로 제공하고 있습니다. 이 시작점을 1로 하도록 설정할 수도 있어 어떤 것이 더 자연스러운지 의견을 묻고 싶네요! 
하지만 1로 설정할 경우 저희가 `responsebody` 에서 `page`필드를 리턴할 때 서버에서 임의로 1을 더해야 하는 작업이 추가적으로 들어가는 살짝의 불편함(?)이 생기긴 합니다.
`int nowPage = historyPageByPetPlantId.getPageable().getPageNumber() + 1;`
위와 같은 형태가 생기게 되는데, 다른 조의 경우 단순히 getPageNumber()를 하면 개발자들이 명시적으로 이해하기 쉬운데, + 1을 하는 행위가 들어가면 헷갈리게 될 여지가 있는 만큼 시스템이 주는 그대로 사용하기로 합의를 했다고 하네용

**Comment by rawfishthelgh** [comment]
1. page, size 정보를 프론트에서 body를 활용해 로직에 활용하는 것이 더 간단하고 통상적으로 그렇게 하는게 대부분이므로 제공한다.
2. 사용자 입장에서 친화적 및 확장성을 고려해 1부터 시작하는 것으로 지정한다.

**Comment by Choi-JJunho** [comment]
해당 주제는 다음 discussion에서 이어서 진행되었으므로 아래 내용을 참고해주세요

- https://github.com/woowacourse-teams/2023-pium/discussions/173
